include/gamepad/recursive_mutex.hpp

```diff
@@ -0,0 +1,27 @@
+#include "pros/apix.h"
+#include "pros/rtos.h"
+#include <system_error>
+
+namespace Gamepad::_impl {
+
+class RecursiveMutex {
+    public:
+        RecursiveMutex()
+            : mutex(pros::c::mutex_recursive_create()) {}
+
+        bool take(std::uint32_t timeout = TIMEOUT_MAX) { return pros::c::mutex_recursive_take(mutex, timeout); }
+
+        void lock() {
+            if (!this->take()) throw std::system_error();
+        }
```

Comment:
```suggestion
        bool lock() {
            return this->take();
        }
```

Return a bool instead, an exception could crash the opcontrol thread

Reply:
Decided to fix this corner case by looping forever while `take()` doesn't succeed since returning `bool` doesn't work nicely with `std::lock_guard`.